try to fix alpha build:

kernel-2.6.27.61/arch/alpha/include/asm/bug.h: In function `unknown_bootoption':
kernel-2.6.27.61/arch/alpha/include/asm/bug.h:13: warning: asm operand 1 probably doesn't match constraints
kernel-2.6.27.61/arch/alpha/include/asm/bug.h:13: warning: asm operand 2 probably doesn't match constraints
kernel-2.6.27.61/arch/alpha/include/asm/bug.h:13: error: impossible constraint in `asm'
kernel-2.6.27.61/arch/alpha/include/asm/bug.h: In function `__BUG':
kernel-2.6.27.61/arch/alpha/include/asm/bug.h:13: warning: asm operand 1 probably doesn't match constraints
kernel-2.6.27.61/arch/alpha/include/asm/bug.h:13: warning: asm operand 2 probably doesn't match constraints
make[1]: *** [init/main.o] Error 1

thus, reverting ed6b9b97f42c091630335bfb71a2931e6f86388b

--- linux-2.6.27/arch/alpha/include/asm/bug.h~	2012-03-16 09:24:29.000000000 +0000
+++ linux-2.6.27/arch/alpha/include/asm/bug.h	2012-03-16 09:24:35.000000000 +0000
@@ -1,24 +1,14 @@
 #ifndef _ALPHA_BUG_H
 #define _ALPHA_BUG_H
 
-#include <linux/linkage.h>
-
 #ifdef CONFIG_BUG
 #include <asm/pal.h>
 
 /* ??? Would be nice to use .gprel32 here, but we can't be sure that the
-   function loaded the GP, so this could fail in modules.  */
-static inline void ATTRIB_NORET __BUG(const char *file, int line)
-{
-	__asm__ __volatile__(
-		"call_pal %0  # bugchk\n\t"
-		".long %1\n\t.8byte %2"
-		       : : "i" (PAL_bugchk), "i"(line), "i"(file));
-	for ( ; ; )
-		;
-}
-
-#define BUG() __BUG(__FILE__, __LINE__)
+ *    function loaded the GP, so this could fail in modules.  */
+#define BUG() \
+	  __asm__ __volatile__("call_pal %0  # bugchk\n\t"".long %1\n\t.8byte %2" \
+			  		       : : "i" (PAL_bugchk), "i"(__LINE__), "i"(__FILE__))
 
 #define HAVE_ARCH_BUG
 #endif
